scripts/import/import_EPMC.pl

```diff
@@ -528,52 +574,78 @@ sub report_summary{
                                          and p1.pmid is not null
                                        ]);
 
-    my $dup2_ext_sth = $dba->dbc->prepare(qq[ select p1.publication_id, p2.publication_id, p2.pmcid
-                                         from publication p1, publication p2
-                                         where p1.pmcid = p2.pmcid
-                                         and p1.publication_id < p2.publication_id
-                                       ]);
-
-   my $dup3_ext_sth = $dba->dbc->prepare(qq[ select p1.publication_id, p2.publication_id, p2.doi
-                                         from publication p1, publication p2
-                                         where p1.doi = p2.doi
-                                         and p1.publication_id < p2.publication_id
-                                       ]);
-
-
-
-
+    # Query is taking a few hours to run
+    # my $dup2_ext_sth = $dba->dbc->prepare(qq[ select p1.publication_id, p2.publication_id, p2.pmcid
+    #                                      from publication p1, publication p2
+    #                                      where p1.pmcid = p2.pmcid
+    #                                      and p1.publication_id < p2.publication_id
+    #                                      and p1.pmcid is not null
+    #                                    ]);
+
+   my $dup2_ext_sth = $dba->dbc->prepare(qq[ SELECT publication_id, pmcid 
```

Comment:
Suggest putting  
- the publication_id in the GROUP BY  
- the pmcid is not NULL in a where clause

Reply:
publication_id is a primary key, so this group by is not going to pick up duplicate pmcids. Suggest:
select pmcid, count(publication_id)  
from publication 
where pmcid IS NOT NULL 
group by pmcid 
having count(publication_id) > 1;